### ISSUE TITLE
SQLite doesn't work

### ISSUE THREAD

**Issue by jaraco** [yougov/pmxbot#26]
I cloned your repository, did a "sudo python setup.py install" then ran "pmxbotweb config.yaml" and I get this:

```
#!python

Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/CherryPy-3.2.3-py2.7.egg/cherrypy/_cprequest.py", line 656, in respond
    response.body = self.handler()
  File "/usr/local/lib/python2.7/dist-packages/CherryPy-3.2.3-py2.7.egg/cherrypy/lib/encoding.py", line 188, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/CherryPy-3.2.3-py2.7.egg/cherrypy/_cpdispatch.py", line 34, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/local/lib/python2.7/dist-packages/pmxbot-1105.4.2dev-py2.7.egg/pmxbot/web/viewer.py", line 284, in default
    for chan in sorted(db.list_channels(), key=lower_case):
  File "/usr/local/lib/python2.7/dist-packages/pmxbot-1105.4.2dev-py2.7.egg/pmxbot/logging.py", line 35, in list_channels
    return self._list_channels()
  File "/usr/local/lib/python2.7/dist-packages/pmxbot-1105.4.2dev-py2.7.egg/pmxbot/logging.py", line 139, in _list_channels
    return (chan[0] for chan in self.db.execute(query).fetchall())
ProgrammingError: SQLite objects created in a thread can only be used in that same thread.The object was created in thread id 140677493421824 and this is thread id 140677333632768
```

----------------------------------------
- Bitbucket: https://bitbucket.org/yougov/pmxbot/issue/26
- Originally reported by: [atlanta800](http://bitbucket.org/atlanta800)
- Originally created at: 2013-02-22T22:17:33.005

**Comment by jaraco** [comment]
I get this as well when using pmxbotweb

----------------------------------------
Original comment by: Chris Jowett

**Comment by jaraco** [comment]
Thanks for reporting this. I acknowledge this is an issue. I do plan to get this fixed sooner than later.

----------------------------------------
Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
Make SQLiteStorage thread-local. Fixes #26.

→ <<cset d1c3bddd96e7>>

----------------------------------------
Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
Released as 1105.4.2. Please give it a try and let me know if you have any other issues.

----------------------------------------
Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
Loaded it up this morning, works great! Thanks.

----------------------------------------
Original comment by: [atlanta800](http://bitbucket.org/atlanta800)
